dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/RetryNTimes.java

```diff
@@ -46,17 +46,17 @@ public void begin() {
 	@Override
 	public void success() {
 		boolean success = false;
-		RetryState rs = state.get();
 		while (!success) {
+			RetryState rs = state.get();
```

Comment:
```
RetryState rs = null;
while(!success) {
    rs = state.get();
....
```

Reply:
why? rs is not being used outside the loop so does it matter?

Reply:
because you re-creating the `RetryState` object in every iteration. There is some performance penalty for that. In the above approach, the object is being re-used in every iteration (not recreated).